movie-app/src/components/switchers-buttons/SwitchersButtons.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useState } from "react";
+import SwitchersButtonWrapper from "./SwitchersButtonWrapper";
+
+const SwitchersButtons = (props: any) => {
+  const [buttons, updateSearchButtons] = useState(props.searchButtonsParams);
+
+  const activateButton = (index: number) => {
+    const newArr = [...buttons];
+    newArr.forEach((item, i) =>
+      i === index ? (item.active = true) : (item.active = false)
```

Comment:
avoid props and state mutation, as it causes unpredictable bugs. Instead use `Array.prototype.map` . Additionally, you win't need to create `newArr` explicitly

Reply:
Fixed